### PR TITLE
Introduce a method to delete all the custom text preferences configured in an organization

### PR DESCRIPTION
**Pull request by UdeshAthukorala**
## Purpose
- Introduce a new method to delete all the custom text preferences configured in an organization

## TODO

- [ ] Bump Frmaework Version

## Related PRs

- https://github.com/wso2/carbon-identity-framework/pull/5030

### CODE REVIEW

File: components/org.wso2.carbon.identity.branding.preference.management.core/pom.xml [hunk 1]
```diff
@@ -204,7 +204,7 @@
                                         <limit implementation="org.jacoco.report.check.Limit">
                                             <counter>COMPLEXITY</counter>
                                             <value>COVEREDRATIO</value>
-                                            <minimum>0.46</minimum>
+                                            <minimum>0.45</minimum>
```
Comment (somindatommy) [inline components/org.wso2.carbon.identity.branding.preference.management.core/pom.xml:207]: Is this intentional?

File: components/org.wso2.carbon.identity.branding.preference.management.core/src/main/java/org/wso2/carbon/identity/branding/preference/management/core/BrandingPreferenceManagerImpl.java [hunk 2]
```diff
@@ -363,6 +365,25 @@ public void deleteCustomText(String type, String name, String screen, String loc
         }
     }
 
+    public void deleteAllCustomText() throws BrandingPreferenceMgtException {
```
Comment (somindatommy) [inline components/org.wso2.carbon.identity.branding.preference.management.core/src/main/java/org/wso2/carbon/identity/branding/preference/management/core/BrandingPreferenceManagerImpl.java:368]: Missing method comment or add the annotation

File: pom.xml [hunk 3]
```diff
@@ -355,7 +355,7 @@
         <pax.logging.api.version>1.10.1</pax.logging.api.version>
 
         <!-- Carbon identity framework version -->
-        <carbon.identity.framework.version>5.20.267</carbon.identity.framework.version>
+        <carbon.identity.framework.version>5.25.389-SNAPSHOT</carbon.identity.framework.version>
```
Comment (somindatommy) [inline pom.xml:358]: Change this before merging

File: components/org.wso2.carbon.identity.branding.preference.management.core/pom.xml (hunk 1 above)
Comment (UdeshAthukorala) [reply components/org.wso2.carbon.identity.branding.preference.management.core/pom.xml:207]: Yes. Add a unit test case for the newly introduced method. But unable to get the previous complexity ratio. It was reduced by a 0.01/.

File: components/org.wso2.carbon.identity.branding.preference.management.core/src/main/java/org/wso2/carbon/identity/branding/preference/management/core/BrandingPreferenceManagerImpl.java (hunk 2 above)
Comment (UdeshAthukorala) [reply components/org.wso2.carbon.identity.branding.preference.management.core/src/main/java/org/wso2/carbon/identity/branding/preference/management/core/BrandingPreferenceManagerImpl.java:369]: Addressed with https://github.com/wso2-extensions/identity-branding-preference-management/pull/20/commits/a5d063ddb596c1f82afbda520f8119eacc2cac6d

**Review by Wathsara — APPROVED** [review]

**Pull request merged**